worker/instancemutater/mutater.go

```diff
@@ -237,13 +237,13 @@ func (m MutaterMachine) processMachineProfileChanges(info *instancemutater.UnitP
 		}
 	}
 
-	verified, err := m.verifyCurrentProfiles(string(info.InstanceId), expectedProfiles)
+	verified, currentProfiles, err := m.verifyCurrentProfiles(string(info.InstanceId), expectedProfiles)
 	if err != nil {
 		return report(errors.Annotatef(err, "%s", m.id))
 	}
 	if verified {
 		m.logger.Infof("no changes necessary to machine-%s lxd profiles (%v)", m.id, expectedProfiles)
-		return report(nil)
+		return report(m.machineApi.SetCharmProfiles(currentProfiles))
```

Comment:
This is changing the behavior juju as the default and juju-model-name profiles are now being written to instanceData where they were not before. We've only been writing the charm profiles, applied, not all profiles. We're also writing profiles to every lxd machine, not just ones with charm specified profiles.

instanceData from the lxd controller:
```
	"machineid" : "0",
	"charm-profiles" : [
		"default",
		"juju-controller"
	]
```

Reply:
> We've only been writing the charm profiles, applied, not all profiles.
verifyCurrentProfiles now calls LXDProfileNames, the value is reflected back to instanceData.
AssignLXDProfiles currently calls LXDProfileNames, the value is reflected back to instanceData.
I don't see the change in behaviour here.

> We're also writing profiles to every lxd machine, not just ones with charm specified profiles.
By this you mean we are reflecting in the instanceData for every machine the profiles that are applied?

Reply:
@hpidcock yes - we used to reflect only charm specific profiles in the instanceData. Now all profiles are being reflected in the instanceData.